packages/headless/src/controllers/insight/result-list/headless-insight-interactive-result.ts

```diff
@@ -0,0 +1,60 @@
+import {pushRecentResult} from '../../../features/recent-results/recent-results-actions';
+import {logDocumentOpen} from '../../../features/result/result-analytics-actions';
+import {InsightEngine} from '../../../insight.index';
+import {Result} from '../../../api/search/search/result';
+import {
+  buildInteractiveResultCore,
+  InteractiveResultCore,
+  InteractiveResultCoreOptions,
+  InteractiveResultCoreProps,
+} from '../../core/interactive-result/headless-core-interactive-result';
+
+export interface InsightInteractiveResultOptions
+  extends InteractiveResultCoreOptions {
+  /**
+   * The query result.
+   */
+  result: Result;
+}
+
+export interface InsightInteractiveResultProps
+  extends InteractiveResultCoreProps {
+  /**
+   * The options for the `InteractiveResult` controller.
+   */
+  options: InsightInteractiveResultOptions;
+}
+
+/**
+ * The `InteractiveResult` controller provides an interface for triggering desirable side effects, such as logging UA events to the Coveo Platform, when a user selects a query result.
+ */
+export interface InsightInteractiveResult extends InteractiveResultCore {}
+
+/**
+ * Creates an `InteractiveResult` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `InteractiveResult` properties.
+ * @returns An `InteractiveResult` controller instance.
+ */
+export function buildInsightInteractiveResult(
```

Comment:
Just spitballing here in case we can prevent from creating a new one but if we need it then we need it.

Reply:
> and would loosening the argument type to something like CoreEngine which both versions extend work?

That would work now, but it would be a risky move. Suppose we change the argument type to `CoreEngine`. It would not be possible for the `buildInteractiveResult` to use anything specific to the `SearchEngine` in the future. And trying to revert the argument back to `SearchEngine` would be a breaking change. Having the `buildInsightInteractiveResult` function can seem redundant at this moment, but it allows us to use use-case-specific engine features later on.

Reply:
Yeah fair enough. Keeping that flexibility is the spirit of this whole plan so you made the right call. Thanks for explaining the reasoning behind it.